src/proxyNetworkProvider.ts

```diff
@@ -122,13 +122,25 @@ export class ProxyNetworkProvider implements INetworkProvider {
         return status;
     }
 
-    async sendTransaction(tx: ITransaction): Promise<string> {
-        let response = await this.doPostGeneric("transaction/send", tx.toSendable());
+    async sendTransaction(tx: ITransaction | ITransactionNext): Promise<string> {
+        if ("toSendable" in tx){
+            const response = await this.doPostGeneric("transactions", tx.toSendable());
+            return response.txHash;
+        }
+
+        const transaction = prepareTransactionNextForBroadcasting(tx);
+        const response = await this.doPostGeneric("transactions", transaction);
         return response.txHash;
     }
 
-    async sendTransactions(txs: ITransaction[]): Promise<string[]> {
-        const data: any = txs.map(tx => tx.toSendable());
+    async sendTransactions(txs: ITransaction[] | ITransactionNext[]): Promise<string[]> {
+        let data: any[];
+        if ("toSendable" in txs[0]){
```

Comment:
This fails if the array is empty. Not a very common case, but I think we should handle it.

I think it's also good to apply the idea you've mentioned on Friday - accept array of `legacy | next` - would result in a nicer code, as well.

Reply:
Now accepting `(ITransaction | ITransactionNext)[]`